runtime/queries/column_numeric_histogram.go

```diff
@@ -196,5 +198,177 @@ func (q *ColumnNumericHistogram) Resolve(ctx context.Context, rt *runtime.Runtim
+			minMaxRow.Close()
+			return err
+		}
+	}
+
+	minMaxRow.Close()
+
+	ticks := 40.0
+	if rng < ticks {
+		ticks = rng
+	}
+	niceResult := NiceAndStep(min, max, ticks)
+	startTick := niceResult[0]
+	endTick := niceResult[1]
+	gap := niceResult[2]
+	if gap < 0.0 {
+		gap = 1 / -gap
+	}
+	bucketCount := int(math.Ceil((endTick - startTick) / gap))
+
+	selectColumn := fmt.Sprintf("%s::DOUBLE", sanitizedColumnName)
+	histogramSQL := fmt.Sprintf(
+		`
+		WITH data_table AS (
+			SELECT %[1]s as %[2]s 
+			FROM %[3]s
+			WHERE %[2]s IS NOT NULL
+		), S AS (
+			SELECT 
+				min(%[2]s) as minVal,
+				max(%[2]s) as maxVal,
+				(max(%[2]s) - min(%[2]s)) as range
+				FROM data_table
+		), values AS (
+			SELECT %[2]s as value from data_table
+			WHERE %[2]s IS NOT NULL
+		), buckets AS (
+			SELECT
+				range as bucket,
+				(range * %[7]f::FLOAT + %[5]f) as low,
+				(range * %[7]f::FLOAT + %7f::FLOAT / 2 + %[5]f) as midpoint,
+				((range + 1) * %[7]f::FLOAT + %[5]f) as high
+			FROM range(0, %[4]d, 1) 
+		),
+		-- bin the values
+		binned_data AS (
+			SELECT 
+				FLOOR(%[4]d::FLOAT * ((value::FLOAT - %[5]f) / %[8]f)) as bucket
+			from values
+		),
+		-- join the bucket set with the binned values to generate the histogram
+		histogram_stage AS (
+			SELECT
+				buckets.bucket,
+				low,
+				high,
+				midpoint,
+				SUM(CASE WHEN binned_data.bucket = buckets.bucket THEN 1 ELSE 0 END) as count
+			FROM buckets
+			LEFT JOIN binned_data ON binned_data.bucket = buckets.bucket
+			GROUP BY buckets.bucket, low, high, midpoint
+			ORDER BY buckets.bucket
+		),
+		-- calculate the right edge, sine in histogram_stage we don't look at the values that
+		-- might be the largest.
+		right_edge AS (
+			SELECT count(*) as c from values WHERE value = %[6]f 
+		)
+		SELECT 
+			bucket,
+			low,
+			high,
+			midpoint,
+		-- fill in the case where we've filtered out the highest value and need to recompute it, otherwise use count.
+			CASE WHEN high = (SELECT max(high) from histogram_stage) THEN count + (select c from right_edge) ELSE count END AS count
+		FROM histogram_stage
+		`,
+		selectColumn,
+		sanitizedColumnName,
+		safeName(q.TableName),
+		bucketCount,
+		startTick,
+		endTick,
+		gap,
+		endTick-startTick,
+	)
+
+	histogramRows, err := olap.Execute(ctx, &drivers.Statement{
+		Query:    histogramSQL,
+		Priority: priority,
+	})
+	if err != nil {
+		return err
+	}
+
+	defer histogramRows.Close()
+
+	histogramBins := make([]*runtimev1.NumericHistogramBins_Bin, 0)
+	for histogramRows.Next() {
+		bin := &runtimev1.NumericHistogramBins_Bin{}
+		err = histogramRows.Scan(&bin.Bucket, &bin.Low, &bin.High, &bin.Midpoint, &bin.Count)
+		if err != nil {
+			return err
+		}
+		histogramBins = append(histogramBins, bin)
+	}
+
+	err = histogramRows.Err()
+	if err != nil {
+		return err
+	}
+
+	q.Result = histogramBins
+
+	return nil
+}
+
+func (q *ColumnNumericHistogram) Resolve(ctx context.Context, rt *runtime.Runtime, instanceID string, priority int) error {
```

Comment:
Move above `calculate...` methods – https://github.com/uber-go/guide/blob/master/style.md#function-grouping-and-ordering